tests/devices/unit_tests/test_aperture_scatterguard.py

```diff
@@ -0,0 +1,52 @@
+from unittest.mock import MagicMock
+
+import pytest
+from ophyd.sim import make_fake_device
+from ophyd.status import Status, StatusBase
+
+from dodal.devices.aperturescatterguard import ApertureScatterguard, InvalidApertureMove
+
+
+@pytest.fixture
+def fake_aperture_scatterguard():
+    FakeApertureScatterguard = make_fake_device(ApertureScatterguard)
+    ap_sg: ApertureScatterguard = FakeApertureScatterguard(name="test_ap_sg")
+    yield ap_sg
+
+
+def test_aperture_scatterguard_throws_error_if_outside_tolerance(
+    fake_aperture_scatterguard: ApertureScatterguard,
+):
+    fake_aperture_scatterguard.aperture.z.motor_resolution.sim_put(0.001)
+    fake_aperture_scatterguard.aperture.z.user_setpoint.sim_put(1)
+    fake_aperture_scatterguard.aperture.z.motor_done_move.sim_put(1)
+
+    with pytest.raises(InvalidApertureMove):
+        fake_aperture_scatterguard._safe_move_within_datacollection_range(
+            0, 0, 1.1, 0, 0
+        )
+
+
+def test_aperture_scatterguard_returns_status_if_within_tolerance(
+    fake_aperture_scatterguard: ApertureScatterguard,
+):
+    fake_aperture_scatterguard.aperture.z.motor_resolution.sim_put(0.001)
+    fake_aperture_scatterguard.aperture.z.user_setpoint.sim_put(1)
+    fake_aperture_scatterguard.aperture.z.motor_done_move.sim_put(1)
+
+    completed_status = Status()
+    completed_status.set_finished()
+
+    mock_set = MagicMock(return_value=Status(done=True, success=True))
```

Comment:
Nit: I think we can just use completed_status as the return value here
`mock_set = MagicMock(return_value=completed_status`)

Reply:
Oops, my bad! I think the better solution is to not have `completed_status` at all, will do that